client/src/BibleEngineClient.ts

```diff
@@ -64,4 +71,31 @@ export class BibleEngineClient {
             throw new Error(`No remote config provided`);
         } else return this.localBibleEngine.getVersions();
     }
+
+    async getBothOfflineAndOnlineVersions() {
```

Comment:
i wonder if this is something that should be part of BibleEngine, since it is app-specific how you handle online/offline (e.g. saving the remote version entity in the local database and syncing). another reason i'd rather recommend to not put it here is that local and remote entities should better not be mixed since they refer to different id-schemas (version id 1 can be different local or remote). if you do that on you app-level bible-service it's another thing however i would output that kind of data from the library itself.

Reply:
I agree. Removed it from the client and put inside the app logic